src/services/payments.js

```diff
@@ -171,13 +145,18 @@ export default class Payments {
 				if (initialSegment === 0 || lastSegment === 0 || (middleSegment > 1 || middleSegment < 0)) {
 					return '';
 				}
-				const initialOutput = '000000'.slice(initialSegment.toString().length) + initialSegment.toString();
-				const middleOutput = middleSegment.toString();
-				const lastOutput = '000000'.slice(lastSegment.toString().length) + lastSegment.toString();
-				return `${initialOutput}${middleOutput}${lastOutput}_IM`;
+				return `${penaltyReference}_IM`;
 			}
 		} else {
-			return `${penaltyReference}_${penaltyType}`;
+			const matches = penaltyReference.match(/^([0-9]{12,13})$/);
+			if (matches === null) {
+				return '';
+			}
+			let retReference = `${penaltyReference}_${penaltyType}`;
+			if (penaltyReference.length < 12 || penaltyReference.length > 13) {
```

Comment:
This if is redundant as the regular expression above already caters for that condition.